gnum4: Does not build with mingwW64

### Describe the bug
When I try and build gnu M4 with the mingwW64 toolchain, I get the following error:
```
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(fopen.o): in function `memcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(setlocale.o): in function `memcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(strerror.o): in function `memcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(strerror.o): in function `sprintf':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(localcharset.o): in function `strcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:228: undefined reference to `__strcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(localcharset.o): in function `sprintf':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(localename.o): in function `strcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:228: undefined reference to `__strcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(nl_langinfo.o): in function `strcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:228: undefined reference to `__strcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(nl_langinfo.o): in function `memmove':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:214: undefined reference to `__memmove_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(nl_langinfo.o): in function `strcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:228: undefined reference to `__strcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:228: undefined reference to `__strcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(nl_langinfo.o): in function `sprintf':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(nl_langinfo.o): in function `memcpy':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(vasnprintf.o): in function `sprintf':
/nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
/nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:2015: m4.exe] Error 1
make[2]: Leaving directory '/build/m4-1.4.19/src'
make[1]: *** [Makefile:2018: all-recursive] Error 1
make[1]: Leaving directory '/build/m4-1.4.19'
make: *** [Makefile:1974: all] Error 2
error: builder for '/nix/store/54sinkm3r9xi2h4vzg48di18wz739yb0-gnum4-x86_64-w64-mingw32-1.4.19.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/string.h:202: undefined reference to `__memcpy_chk'
       > /nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: ../lib/libm4.a(vasnprintf.o): in function `sprintf':
       > /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
       > /nix/store/adb9g876bb4yz4bh4maps9gnzq8d5mkz-x86_64-w64-mingw32-binutils-2.35.2/bin/x86_64-w64-mingw32-ld: /nix/store/w6fy7c8qj11s2k06mbdhj3cq580dydnl-x86_64-w64-mingw32-stage-final-gcc-debug-10.3.0/x86_64-w64-mingw32/sys-include/stdio.h:386: undefined reference to `__chk_fail'
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:2015: m4.exe] Error 1
       > make[2]: Leaving directory '/build/m4-1.4.19/src'
       > make[1]: *** [Makefile:2018: all-recursive] Error 1
       > make[1]: Leaving directory '/build/m4-1.4.19'
       > make: *** [Makefile:1974: all] Error 2
       For full logs, run 'nix log /nix/store/54sinkm3r9xi2h4vzg48di18wz739yb0-gnum4-x86_64-w64-mingw32-1.4.19.drv'.
```

### Steps To Reproduce
Run the command `nix-build '<nixpkgs>' -A pkgsCross.mingwW64.gnum4`

### Additional context
The override:
```nix
gnum4 = super.gnum4.overrideAttrs (oldAttrs: rec {
  configureFlags = oldAttrs.configureFlags ++ [ "CFLAGS=-fno-stack-protector" ];
});
```
makes it build, but I don't know if it's the correct fix.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@toonn 
@KAction 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.5, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-21.11.337381.d4191fe35cb, nixos-unstable-22.05pre365028.710fed5a248"`
 - channels(rendal): `"home-manager-21.11, nixos-unstable-22.05pre356435.7f9b6e2babf"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

--- a/pkgs/development/tools/misc/gnum4/default.nix
+++ b/pkgs/development/tools/misc/gnum4/default.nix
@@ -16,7 +16,7 @@ stdenv.mkDerivation rec {
 
   doCheck = false;
 
-  configureFlags = [ "--with-syscmd-shell=${stdenv.shell}" ];
+  configureFlags = [ "--with-syscmd-shell=${stdenv.shell}" ] ++ lib.optional stdenv.hostPlatform.isMinGW [ "CFLAGS=-fno-stack-protector" ];
 
   meta = {
     description = "GNU M4, a macro processor";

--- a/pkgs/development/tools/misc/gnum4/default.nix
+++ b/pkgs/development/tools/misc/gnum4/default.nix
@@ -16,7 +16,8 @@ stdenv.mkDerivation rec {
 
   doCheck = false;
 
-  configureFlags = [ "--with-syscmd-shell=${stdenv.shell}" ] ++ lib.optional stdenv.hostPlatform.isMinGW [ "CFLAGS=-fno-stack-protector" ];
+  configureFlags = [ "--with-syscmd-shell=${stdenv.shell}" ]
+    ++ lib.optional stdenv.hostPlatform.isMinGW "CFLAGS=-fno-stack-protector";
 
   meta = {
     description = "GNU M4, a macro processor";

